pkg/new_storage/cos.go

```diff
@@ -77,10 +77,12 @@ func (c *COS) DeleteFile(key string) error {
 	return err
 }
 
-func (c *COS) Walk(cosPath string, recursuve bool, process func(RemoteFile) error) error {
+func (c *COS) Walk(cosPath string, recursive bool, process func(RemoteFile) error) error {
 	prefix := path.Join(c.Config.Path, cosPath)
 	delimiter := ""
-	if !recursuve {
+	// We force recursive equals true.
+	recursive = true
```

Comment:
Maybe better will just always use `delimiter := "" `

And complete remove recursive = true and `if !recusive { delimiter = "/" }` ?

Reply:
> Maybe better will just always use `delimiter := "" `
> 
> And complete remove `recursive = true` and `if !recusive { delimiter = "/" }` ?

For the reason that Go does not support default value for parameters, we can not remove this "recursive". Maybe, we can add another function that pass the true to next invocation. That's the rule of COS(Close for modification and Open for Extension) Rule.